File: Java/NoiseFilters3Methods.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class NoiseFilters3Methods {

	public static void main(String[] args) {

		String dataFileName;
		String maskFileName;
		String averageFilterFileName;
		String medianFilterFileName;
		String gaussianFilterFileName;

		if (args.length != 5) {
			System.out.println("Invalid number of arguments.");
			System.out.println(
					"There should be data file, mask file, averageFilterOutput file, gaussianFilterOutputFile, amd medianFilterOutputFile.");
			System.exit(0);
		}
		try {

			dataFileName = args[0];
			maskFileName = args[1];
			averageFilterFileName = args[2];
			medianFilterFileName = args[4];
			gaussianFilterFileName = args[3];

			Scanner data = new Scanner(new File(dataFileName));
			Scanner mask = new Scanner(new File(maskFileName));
			FileWriter averageFilter = new FileWriter(new File(averageFilterFileName));
			FileWriter medianFilter = new FileWriter(new File(medianFilterFileName));
			FileWriter gaussianFilter = new FileWriter(new File(gaussianFilterFileName));

			ImageProcessing image = new ImageProcessing(data, mask);
			image.loadImage(data, image.mirrorFramedArray);
			image.loadMask(mask, image.maskArray);
			image.mirrorFraming(image.mirrorFramedArray);

			image.computeAverageFilterImage(image.averageFilterArray);
			image.outputArrayToFile(image.averageFilterArray, averageFilter);

			image.computeMedianFilterImage(image.medianFilterArray);
			image.outputArrayToFile(image.medianFilterArray, medianFilter);

			image.computeGaussianFilterImage(image.gaussianFilterArray);
			image.outputArrayToFile(image.gaussianFilterArray, gaussianFilter);

			data.close();
			mask.close();
			averageFilter.close();
			medianFilter.close();
			gaussianFilter.close();
		} catch (IOException e) {
			e.printStackTrace();
		}
	}

}

File: C++/NoiseFilters3Methods.cpp
#include <iostream>
#include <fstream>
#include <string>

#include "ImageProcessing.h"

using namespace std;

int main(int argc, char** argv) {
	string dataFileName;
	string maskFileName;
	string averageFilterFileName;
	string medianFilterFileName;
	string gaussianFilterFileName;

	if (argc != 6) {
		cout << "Invalid number of arguments."<< endl;
		cout << "There should be data file, mask file, averageFilterOutput file, gaussianFilterOutputFile, amd medianFilterOutputFile." << endl;
		exit(0);
	}
	try {

		dataFileName = argv[1];
		maskFileName = argv[2];
		averageFilterFileName = argv[3];
		medianFilterFileName = argv[5];
		gaussianFilterFileName = argv[4];

		ifstream data;
		ifstream mask;
		ofstream averageFilter;
		ofstream medianFilter;
		ofstream gaussianFilter;

		data.open(dataFileName);
		mask.open(maskFileName);
		averageFilter.open(averageFilterFileName);
		medianFilter.open(medianFilterFileName);
		gaussianFilter.open(gaussianFilterFileName);

		ImageProcessing* image = new ImageProcessing(data, mask);
		image->loadImage(data, image->mirrorFramedArray);
		image->loadMask(mask, image->maskArray);
		image->mirrorFraming(image->mirrorFramedArray);

		image->computeAverageFilterImage(image->averageFilterArray);
		image->outputArrayToFile(image->averageFilterArray, averageFilter);

		image->computeMedianFilterImage(image->medianFilterArray);
		image->outputArrayToFile(image->medianFilterArray, medianFilter);

		image->computeGaussianFilterImage(image->gaussianFilterArray);
		image->outputArrayToFile(image->gaussianFilterArray, gaussianFilter);


		data.close();
		mask.close();
		averageFilter.close();
		medianFilter.close();
		gaussianFilter.close();
	}
	catch (exception e) {
		cout << "There is an error: " << e.what();
	}
	return 0;
}